### ISSUE TITLE
How to use CDN import correctly

### ISSUE THREAD

**Issue by China-xiaoFang** [prazdevs/pinia-plugin-persistedstate#348]
### Are you using Nuxt?

- [ ] Check this box if you encountered the issue using Nuxt and the included module.

### Describe the bug

When I tried to use `pinia-plugin-persistedstate` in `Vue3` + `Vite` + `TypeScript` environment, I imported the library through CDN, and the browser console reported an error
```bash
Uncaught ReferenceError: module is not defined
at index.js:26:8
(Anonymous) @ index.js:26
index-nOvpWTLJ.js:857 Uncaught (in promise) ReferenceError: piniaPluginPersistedstate is not defined
at nt (index-nOvpWTLJ.js:857:42003)
```
![image](https://github.com/user-attachments/assets/374a7776-963b-463c-9190-34f6d5d38cec)
![image](https://github.com/user-attachments/assets/c9c8b5ba-8091-4de5-a434-edf8b85d2a95)

I think this should be a problem of not handling umd format build during build.
Because it relies on third-party dependencies, `deep-pick-omit`, `defu`, `destr` Although `pnpm` will automatically load these third-party dependencies in ES mode, they will not be automatically loaded in CDN mode, so these third-party dependencies cannot be excluded when building `umd` format js files. Otherwise, the CDN reference may not be able to find the third-party dependencies or may report an import error.

I wanted to modify the build configuration myself by cloning the repository, but I found that the cloned project seemed to lack some necessary files, such as `./.nuxt/tsconfig.json`.
![image](https://github.com/user-attachments/assets/99e2642b-f195-446f-9012-2688a80ff81a)
![image](https://github.com/user-attachments/assets/7fd998fe-e284-4eff-af58-b5343fb3da5f)

How to reproduce the smallest demo provided:
`pnpm i`
The development environment is normal
`pnpm dev`
There will be problems with previewing after packaging, because it is referenced through CDN at this time
`pnpm build`
`pnpm preview`
Open the browser console and you will find errors.

### Reproduction

https://stackblitz.com/edit/pinia-plugin-persistedstate-cdn-error

### System Info

```Shell
System:
    OS: Windows 11 10.0.22631
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12700
    Memory: 9.70 GB / 31.79 GB
  Binaries:
    Node: 20.15.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.7.0 - C:\Program Files\nodejs\npm.CMD
    pnpm: 9.11.0 - C:\Program Files\nodejs\pnpm.CMD
  Browsers:
    Edge: Chromium (129.0.2792.65)
    Internet Explorer: 11.0.22621.3527
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/prazdevs/pinia-plugin-persistedstate/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/prazdevs/pinia-plugin-persistedstate/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by prazdevs** [comment]
Hi, from `v4.1.2` you should import from `https://cdn.jsdelivr.net/npm/pinia-plugin-persistedstate@4.1.2/dist/index.global.js` and it should work. 

Let me know if that fixes the issue!

**Comment by China-xiaoFang** [comment]
![image](https://github.com/user-attachments/assets/5befc58e-1f74-407e-9c64-af12a1f71995)

Hello, thank you for taking the time to deal with this issue.
However, after my test, as shown in the figure, I still can't find the global var name, but it may be that my global var name is used incorrectly.
Maybe the `iife` build does not support this operation.
Generally, we build libraries through the `umd` mode, so I think you should add the `umd` mode build and specify the global var name, which may solve this problem.
In the smallest demo provided before, I have updated and reproduced this problem.

**Comment by prazdevs** [comment]
I will try to use Rollup for bundling when I get some time and see if i can make a "working" umd build. 😌

**Comment by prazdevs** [comment]
There should now be 2 "global" CDN imports available. 
- `https://cdn.jsdelivr.net/npm/pinia-plugin-persistedstate@4.1.3/dist/index.global.js` is an `iife` bundle
- `https://cdn.jsdelivr.net/npm/pinia-plugin-persistedstate@4.1.3/dist/index.umd.js` is an `umd` bundle

I don't really know/use those kind of bundles, so let me know if the `umd` one works for you, or if you expect/need something else. 😄

**Comment by China-xiaoFang** [comment]
Hello, after my test, I think this problem should be solved.
It seems that there is no problem for packages in `umd` or `iife` format.
I compared the `index.global.js` files built by `4.1.2` and `4.1.3`, and found that `4.1.2` lacked the global var piniaPluginPersistedstate name, which caused this problem.
It was solved in `4.1.3`.
Finally, thank you again for your reply. Thank you!

**Comment by China-xiaoFang** [comment]
Hello, after my test, I think this problem should be solved.
It seems that there is no problem for packages in `umd` or `iife` format.
I compared the `index.global.js` files built by `4.1.2` and `4.1.3`, and found that `4.1.2` lacked the global var piniaPluginPersistedstate name, which caused this problem.
It was solved in `4.1.3`.
Finally, thank you again for your reply. Thank you!
